platform/server/plugins/PluginSystem.ts

```diff
@@ -24,7 +24,7 @@ function getSortedPluginNames(plugins: Map<PluginName, Plugin<any, any>>) {
 }
 
 export class PluginSystem {
-  private readonly plugins = new Map<PluginName, Plugin<{}, {}>>();
```

Comment:
This and the one above started failing because of stricter variance checking. I think this is okey, as it doesn't cause any problems, but I'll think about potential ideas for the future.